### DISCUSSION TITLE
Using Searchable tool using snoflake syncer but data failing to load due to datatype mismtach

### DISCUSSION THREAD

**Discussion by kazi-ps** [thoughtspot/cs_tools#163] [Q&A] [answered]
Hi All

I am trying to run 
`cs_tools tools searchable metadata --syncer snowflake://D:/export/sf_def.toml --config prod`

The tables are created in Snowflake 'ts_cluster', 'ts_column_synonym', 'ts_data_source', 'ts_dependent_object', 'ts_group', 'ts_group_privilege', 'ts_metadata_column', 'ts_metadata_object', 'ts_org', 'ts_sharing_access', 'ts_tag', 'ts_tagged_object', 'ts_user', 'ts_xref_org', 'ts_xref_principal'

But while inserting data from temp csv to SF it is giving error
`raise error_class( sqlalchemy.exc.ProgrammingError: (snowflake.connector.errors.ProgrammingError) 100035 (22007): Timestamp '2023-05-12 18:44:47+00:00' is not recognized File 'tmpsy0ffij6_ts_user.csv.gz', line 1, character 136 Row 1, column "TS_USER"["CREATED":7] If you would like to continue loading when an error is encountered, use other values such as 'SKIP_FILE' or 'CONTINUE' for the ON_ERROR option. For more information on loading options, please run 'info loading_data' in a SQL client. [SQL: COPY INTO ts_user FROM @DB.SCHEMA.TMP_STAGE_ts_user_6cfb1 ON_ERROR = ABORT_STATEMENT PURGE = TRUE ] (Background on this error at: [https://sqlalche.me/e/20/f405)`](https://sqlalche.me/e/20/f405)%60)`

Given that the CREATED column in ts_users is of type TIMESTAMP_NTZ(9), it means that the timestamp should not include time zone information. However, CSV file contains timestamps with a time zone offset (+00:00). 

I tried changing the table definition to TIMESTAMP_TZ but it still fails with same message?

Is this a bug?

**Comment by boonhapus** [comment]
@kazi-ps for your syncer definition, do you specify `load_strategy` at all? Can you try using `load_strategy=TRUNCATE` on this one?

**Reply by kazi-ps** [reply]
@boonhapus  yes, I have this defined in syncer - load_strategy='truncate'

**Reply by kazi-ps** [reply, marked as answer]
@boonhapus - I updated syncer.py file where the file format is defined and added this option 
`TIMESTAMP_FORMAT = 'YYYY-MM-DD HH24:MI:SS TZH:TZM'`
This resolved the issue
